test/suites/completions.sh

```diff
@@ -0,0 +1,330 @@
+    [ "$(complete delete --force c)" = 'c1,c2' ]
+    [ "$(complete delete --force localhost:c)" = 'localhost:c1,localhost:c2' ]
+    [ "$(complete exec '')" = 'c2,localhost:' ] # Exec should show only instance server remotes and not the default, c1 should not be shown because it is not running.
+    [ "$(complete exec l)" = 'localhost:' ]
+    [ "$(complete exec c)" = 'c2' ]
+    [ "$(complete exec localhost:c)" = 'localhost:c2' ]
+    [ "$(complete export '')" = 'c1,c2,localhost:' ] # Export should show only instance server remotes and not the default
+    [ "$(complete export l)" = 'localhost:' ]
+    [ "$(complete export c)" = 'c1,c2' ]
+
+    (
+      import_wd=$(mktemp -d -p "${TEST_DIR}" XXX)
+      # shellcheck disable=SC2164
+      # (tests are run with set -e)
+      cd "${import_wd}"
+      mkdir foo
+      touch bar.txt fizz.tar buzz.tar.gz bazz.tar.xz
+      [ "$(complete import '')" = 'buzz.tar.gz,foo/,localhost:' ] # Correct file extensions, directories, and non-default remotes
+      [ "$(complete import localhost: '')" = 'buzz.tar.gz,foo/' ]
+      rm -rf "${import_wd}"
+    )
+
+    [ "$(complete info '')" = 'c1,c2,localhost:' ] # Info should show only instance server remotes and not the default
+    [ "$(complete info l)" = 'localhost:' ]
+    [ "$(complete info c)" = 'c1,c2' ]
+    testimage_fingerprint="$(lxc query /1.0/images/aliases/testimage | jq -r .target)"
+    testimage_fingerprint_prefix="$(echo "${testimage_fingerprint}" | cut -c1-12)"
+    [ "$(complete init '')" = "${testimage_fingerprint_prefix},images:,localhost:,testimage,ubuntu-daily:,ubuntu-minimal-daily:,ubuntu-minimal:,ubuntu:" ] # Init should show images from the default remote, and all image remotes
+    [ "$(complete init l)" = 'localhost:' ]
+    [ "$(complete launch '')" = "${testimage_fingerprint_prefix},images:,localhost:,testimage,ubuntu-daily:,ubuntu-minimal-daily:,ubuntu-minimal:,ubuntu:" ] # Launch should show images from the default remote, and all image remotes
+    [ "$(complete launch l)" = 'localhost:' ]
+    [ "$(complete monitor '')" = 'localhost:' ] # Monitor should show only instance server remotes and not the default
+    [ "$(complete monitor l)" = 'localhost:' ]
+    [ "$(complete move '')" = 'c1,c2,localhost:' ] # Move should show only instance server remotes and not the default
+    [ "$(complete move l)" = 'localhost:' ]
+    [ "$(complete move c)" = 'c1,c2' ]
+    [ "$(complete pause '')" = 'c2,localhost:' ] # Only c2 running
+    [ "$(complete publish '')" = 'c1,c2,localhost:' ]
+    [ "$(complete publish l)" = 'localhost:' ]
+    [ "$(complete publish c)" = 'c1,c2' ]
+    [ "$(complete publish c1/s)" = 'c1/snap0' ]
+    [ "$(complete publish c1/)" = 'c1/snap0' ]
+    [ "$(complete rebuild '')" = "${testimage_fingerprint_prefix},images:,localhost:,testimage,ubuntu-daily:,ubuntu-minimal-daily:,ubuntu-minimal:,ubuntu:" ]
+    [ "$(complete rebuild u)" = "ubuntu-daily:,ubuntu-minimal-daily:,ubuntu-minimal:,ubuntu:" ]
```

Comment:
Can we actually rebuild a given image fingerprint?

Simplestreams remote should also be filtered out IMHO:

```
$ lxc rebuild ubuntu:noble --empty
Error: The remote isn't a private LXD server
```

Reply:
oh, my bad, I didn't understand how rebuild worked. You indeed need to provide a remote for the source of the image to use :/